config/setupMPI.cmake

```diff
@@ -572,30 +572,35 @@ macro( setupMPILibrariesWindows )
    #     del msmpi.def
    #     copy libmsmpi.a %MSMPI_LIB32%/libmsmpi.a
 
-   if( EXISTS ${CMAKE_Fortran_COMPILER} )
-   # only do this if we are in a CMakeAddFortranSubdirectory directive when the
-   # main Generator is Visual Studio and the Fortran subdirectory uses gfortran
-   # with Makefiles.
-   if( "${MPI_Fortran_LIBRARIES}none" STREQUAL "none" OR
-       "${MPI_Fortran_LIBRARIES}" MATCHES "msmpi.lib" )
-     # should be located in ENV{PATH} at c:/Program Files/Microsoft MPI/Bin/
-     find_library( MPI_gfortran_LIBRARIES msmpi )
-     set( MPI_Fortran_LIBRARIES ${MPI_gfortran_LIBRARIES} CACHE FILEPATH
-          "msmpi for gfortran" FORCE )
-        
-     # Windows systems with dll libraries.
-     add_library( MPI::MPI_Fortran SHARED IMPORTED )
+   if( WIN32 AND EXISTS "${CMAKE_Fortran_COMPILER}" AND
+       TARGET MPI::MPI_Fortran )
+
+     # only do this if we are in a CMakeAddFortranSubdirectory directive
+     # when the main Generator is Visual Studio and the Fortran subdirectory
+     # uses gfortran with Makefiles.
+
+     # MS-MPI has an architecture specific include directory that
+     # FindMPI.cmake doesn't seem to pickup correctly.  Add it here.
+     get_target_property(mpilibdir MPI::MPI_Fortran
+       INTERFACE_LINK_LIBRARIES)
+     get_target_property(mpiincdir MPI::MPI_Fortran
+       INTERFACE_INCLUDE_DIRECTORIES)
+     foreach( arch x86 x64 )
+       string(FIND "${mpilibdir}" "lib/${arch}" found_lib_arch)
+       string(FIND "${mpiincdir}" "include/${arch}" found_inc_arch)
+       if( ${found_lib_arch} GREATER 0 AND
+         ${found_inc_arch} LESS 0 )
+         if( IS_DIRECTORY "${mpiincdir}/${arch}")
+           list(APPEND mpiincdir "${mpiincdir}/${arch}")
+           set_target_properties(MPI::MPI_Fortran
```

Comment:
Since the `mpiincdir` list is accumulated here, can `set_target_properties` be invoked outside the `foreach` loop?

Reply:
Yes! Nice catch. I'll update this PR shortly.